ext/zend_test/test_arginfo.h

```diff
@@ -131,3 +146,100 @@ static const zend_function_entry class_ZendTestNS2_Foo_methods[] = {
 	ZEND_ME(ZendTestNS2_Foo, method, arginfo_class_ZendTestNS2_Foo_method, ZEND_ACC_PUBLIC)
 	ZEND_FE_END
 };
+
+zend_class_entry *register_class__ZendTestInterface()
+{
+	zend_class_entry ce, *class_entry;
+
+	INIT_CLASS_ENTRY(ce, "_ZendTestInterface", class__ZendTestInterface_methods);
+	class_entry = zend_register_internal_interface(&ce);
+
+	return class_entry;
+}
+
+zend_class_entry *register_class__ZendTestClass(zend_class_entry *class_entry__ZendTestInterface, zend_class_entry *class_entry_stdClass)
+{
+	zend_class_entry ce, *class_entry;
+
+	INIT_CLASS_ENTRY(ce, "_ZendTestClass", class__ZendTestClass_methods);
+	class_entry = zend_register_internal_class_ex(&ce, NULL);
+	zend_class_implements(class_entry, 1, class_entry__ZendTestInterface);
+	zend_register_class_alias("_ZendTestClassAlias", class_entry);
+
+	zend_declare_property_null(class_entry, "_StaticProp", sizeof("_StaticProp") - 1, ZEND_ACC_PUBLIC|ZEND_ACC_STATIC);
+
+	zval property_staticIntProp_default_value;
+	ZVAL_LONG(&property_staticIntProp_default_value, 123);
+	zend_string *property_staticIntProp_name = zend_string_init("staticIntProp", sizeof("staticIntProp") - 1, 1);
+	zend_declare_typed_property(class_entry, property_staticIntProp_name, &property_staticIntProp_default_value, ZEND_ACC_PUBLIC|ZEND_ACC_STATIC, NULL, (zend_type) ZEND_TYPE_INIT_MASK(MAY_BE_LONG));
+	zend_string_release(property_staticIntProp_name);
+
+	zval property_intProp_default_value;
+	ZVAL_LONG(&property_intProp_default_value, 123);
+	zend_string *property_intProp_name = zend_string_init("intProp", sizeof("intProp") - 1, 1);
+	zend_declare_typed_property(class_entry, property_intProp_name, &property_intProp_default_value, ZEND_ACC_PUBLIC, NULL, (zend_type) ZEND_TYPE_INIT_MASK(MAY_BE_LONG));
+	zend_string_release(property_intProp_name);
+
+	zval property_classProp_default_value;
+	ZVAL_NULL(&property_classProp_default_value);
+	zend_string *property_classProp_name = zend_string_init("classProp", sizeof("classProp") - 1, 1);
+	zend_declare_typed_property(class_entry, property_classProp_name, &property_classProp_default_value, ZEND_ACC_PUBLIC, NULL, (zend_type) ZEND_TYPE_INIT_CE(class_entry_stdClass, 1, 0));
```

Comment:
I've just noticed that besides `ZEND_TYPE_INIT_CE`, there is also a `ZEND_TYPE_INIT_CLASS` function. Which one should I prefer?

Reply:
Both can be used. INIT_CLASS is resolved to INIT_CE during startup. Using INIT_CLASS might be more convenient, in that it does not require passing through the class entry to the register function.

Reply:
Then `ZEND_TYPE_INIT_CLASS` is the clear winner :)

Reply:
I tried this approach, but there is again an issue with `Throwable`. The assertion failure says that `php: /home/vsts/work/1/s/Zend/zend.c:975: resolve_type_name: Assertion `ce && ce->type == 1' failed.`, so it seems that it's a similar ordering issue that you fixed recently.

Reply:
I don't really see why that would happen. Can you push your attempt?

Reply:
Ah, I see that you already did ^^ A bit stumped here... The class is in the class_table and we're looking it up with the right name, and yet it is not being found.

Reply:
Yeds :) I recently realized that INIT_CLASS needs a `zend_string` so I fixed it with my latest commit. And after locally commenting out the registration of `Error::$previous` and `Exception::$previous`, the build was successful.

Reply:
This turned out to be a simple problem: The length of the string was wrong.

Reply:
Oh, shit.